mdbackup/storage/backblaze.py

```diff
@@ -30,17 +31,18 @@ class B2Storage(AbstractStorage[str]):
 
     def __init__(self, config):
         self.__log = logging.getLogger(__name__)
-        self.__b2 = B2(key_id=config['keyId'], application_key=config['appKey'])
-        self.__bucket: B2Bucket = self.__b2.buckets.get(config['bucket'])
-        if self.__bucket is not None:
-            raise Exception(f'The bucket {config['bucket']} does not exist or could not connect to it. Check configuration...')
+        self.__info = InMemoryAccountInfo()
+        self.__b2_api = b2_api = B2Api(self.__info)
+        self.__b2 = b2_api.authorize_account("production", application_key_id=config['keyId'], application_key=config['appKey'])
+        self.__bucket: bucket = self.__b2_api.get_bucket_by_name(config['bucket'])
         self.__password: str = config.get('password')
         self.__pre = config.backups_path if not config.backups_path.endswith('/') else config.backups_path[:-1]
 
     def list_directory(self, path: Union[str, Path, str]) -> List[str]:
         path = path if isinstance(path, str) else str(path)
-        return [item.file_name for item in self.__bucket.files.all(include_hidden=True)
-                if item.file_name.startswith(self.__pre + '/' + path)]
+        return [f['fileName'] for f in bucket.list_file_names()['files']]
+                if path.startswith('/'):
```

Comment:
It should be before the return. Also this is a syntax error in Python because of the indentation.